agentcfg/fetch.go

```diff
@@ -91,13 +81,28 @@ func (f *Fetcher) request(r io.Reader, err error) ([]byte, error) {
 	}
 
 	result, err := ioutil.ReadAll(resp.Body)
-	if resp.StatusCode == http.StatusMultipleChoices {
-		return nil, errors.Wrap(errors.New(string(result)), ErrMsgMultipleChoices)
-	} else if resp.StatusCode > http.StatusMultipleChoices {
+	if resp.StatusCode >= http.StatusBadRequest {
 		return nil, errors.New(string(result))
 	}
 	if err != nil {
 		return nil, errors.Wrap(err, ErrMsgReadKibanaResponse)
 	}
 	return result, nil
 }
+
+func sanitize(isRum bool, result Result) Result {
+	if !isRum {
+		return result
+	}
+	hasRumData := utility.Contains(result.Source.Agent, RumAgent) || result.Source.Agent == ""
+	if !hasRumData {
+		return zeroResult()
+	}
+	settings := Settings{}
+	for k, v := range result.Source.Settings {
+		if utility.Contains(k, RumSettings) {
+			settings[k] = v
+		}
+	}
```

Comment:
nit: why not deleting settings that should not be contained, instead of creating a new settings object?

Reply:
because i don't want to modify the argument